lib/creator.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class Creator
+  def initialize(user_interface, database)
+    @database = database
+    @user_interface = user_interface
+  end
+
+  def run
+    loop do
+      contact_details = user_interface.ask_for_fields
+
+      database.create(contact_details)
+      user_interface.display(contact_details)
+      break if user_interface.add_another_contact? == false
```

Comment:
it's an antipattern to compare the result of a boolean method with a boolean, the boolean method is already returning a boolean.

If what you want is the opposite boolean, you can use `unless` instead of `if`:

```ruby
break unless user_interface.add_another_contact?
```